apps/admin-gui/src/app/vos/pages/vo-detail-page/vo-groups/vo-groups.component.ts

```diff
@@ -1,43 +1,92 @@
 import { DeleteGroupDialogComponent } from '../../../../shared/components/dialogs/delete-group-dialog/delete-group-dialog.component';
 import { SelectionModel } from '@angular/cdk/collections';
 import { MoveGroupDialogComponent } from '../../../../shared/components/dialogs/move-group-dialog/move-group-dialog.component';
 import { getDefaultDialogConfig } from '@perun-web-apps/perun/utils';
-import { Group, GroupsManagerService, RichGroup, Vo } from '@perun-web-apps/perun/openapi';
-import { GroupFlatNode } from '@perun-web-apps/perun/models';
+import {
+  Group,
+  GroupsManagerService,
+  GroupsOrderColumn,
+  PaginatedRichGroups,
+  RichGroup,
+  Vo,
+} from '@perun-web-apps/perun/openapi';
+import { GroupFlatNode, PageQuery } from '@perun-web-apps/perun/models';
 import { TABLE_VO_GROUPS } from '@perun-web-apps/config/table-config';
 import { Urns } from '@perun-web-apps/perun/urns';
 import { EntityStorageService, GuiAuthResolver } from '@perun-web-apps/perun/services';
 import { MatSlideToggle } from '@angular/material/slide-toggle';
-import { GroupsListComponent, GroupsTreeComponent } from '@perun-web-apps/perun/components';
+import { BehaviorSubject, merge, Observable, Subject } from 'rxjs';
+import { map, mapTo, startWith, switchMap, tap } from 'rxjs/operators';
 
 @Component({
   selector: 'app-vo-groups',
   templateUrl: './vo-groups.component.html',
   styleUrls: ['./vo-groups.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class VoGroupsComponent implements OnInit {
   static id = 'VoGroupsComponent';
   @HostBinding('class.router-component') true;
-  @ViewChild('toggle', { static: true })
-  toggle: MatSlideToggle;
-  @ViewChild('tree', {})
-  tree: GroupsTreeComponent;
-  @ViewChild('list', {})
-  list: GroupsListComponent;
+  @ViewChild('toggle', { static: true }) toggle: MatSlideToggle;
   vo: Vo;
   groups: RichGroup[] = [];
+  nextPage = new Subject<PageQuery>();
+  groupPage$: Observable<PaginatedRichGroups> = this.nextPage.pipe(
+    switchMap((pageQuery) =>
+      this.groupService.getGroupsPage({
+        vo: this.vo.id,
+        attrNames: this.attrNames,
+        query: {
+          order: pageQuery.order,
+          pageSize: pageQuery.pageSize,
+          offset: pageQuery.offset,
+          searchString: pageQuery.searchString,
+          sortColumn: pageQuery.sortColumn as GroupsOrderColumn,
+        },
+      })
+    ),
+    // 'Tapping' is generally a last resort
+    tap((page) => {
+      this.groups = page.data;
+      this.setAuthRights();
+      this.selected.clear();
+      setTimeout(() => this.loadingSubject$.next(false), 200);
+    }),
+    startWith({ data: [], totalCount: 0, offset: 0, pageSize: 0 })
+  );
+
   showGroupList = false;
   selected = new SelectionModel<RichGroup>(true, []);
-  loading: boolean;
+  loadingSubject$ = new BehaviorSubject(false);
+  loading$ = merge(this.loadingSubject$, this.nextPage.pipe(mapTo(true)));
```

Comment:
```suggestion
  loading$: Observable<boolean> = merge(this.loadingSubject$, this.nextPage.pipe(mapTo(true)));
```
mapTo(...) - deprecated, use map() instead of this. Also if I want to refresh the vo-groups page, on this page appears the spinner, but this spinner will never be hidden again and you can see infinity loading.